src/sentry/snuba/subscriptions.py

```diff
@@ -193,14 +193,12 @@ def delete_snuba_subscription(subscription):
         _delete_from_snuba(QueryDatasets(subscription.dataset), subscription.subscription_id)
 
 
-def _create_in_snuba(
-    project, dataset, query, aggregation, time_window, resolution, environment_names
-):
+def _create_in_snuba(project, dataset, query, aggregation, time_window, resolution, environments):
     conditions = resolve_discover_aliases({"conditions": get_filter(query).conditions})[0][
         "conditions"
     ]
-    if environment_names:
-        conditions.append(["environment", "IN", environment_names])
+    if environments:
```

Comment:
I'd suggest coercing this to a list first before checking truthiness unless something changed in modern Django.

Without doing `environments = list(environments)` first, it seems you pass this in as a QuerySet still, which means this will evaluate a query here for the `bool(...)` then you're ultimately iterating it again later.

I'm actually unsure of the exact behaviors here and what queries are performed, if it it gets cached after the `bool()` check. I prefer to be explicit and just coerce it to a list first to make sure we're doing the most efficient thing.

Reply:
I guess according to this: https://docs.djangoproject.com/en/1.11/ref/models/querysets/#when-querysets-are-evaluated

it seems it'd be fine. For some reason i thought `bool(..)` ended up yielding a COUNT(*) query orsomething similar so it didn't cache anything.

Reply:
Doesn't hurt to cast it, just going to do so.